src/AnalysisServices/AnalysisServices.Dataplane/Commands/Restart-AzureAsInstance.cs

```diff
@@ -12,140 +12,35 @@
     /// Cmdlet to log into an Analysis Services environment
     /// </summary>
-    [Cmdlet("Restart", ResourceManager.Common.AzureRMConstants.AzurePrefix + "AnalysisServicesInstance", SupportsShouldProcess=true)]
+    [Cmdlet("Restart", ResourceManager.Common.AzureRMConstants.AzurePrefix + "AnalysisServicesInstance", SupportsShouldProcess = true)]
     [Alias("Restart-AzureAsInstance", "Restart-AzAsInstance")]
     [OutputType(typeof(bool))]
-    public class RestartAzureAnalysisServer : AzurePSCmdlet
+    public class RestartAzureAnalysisServer : AsAzureDataplaneCmdletBase
     {
-        private string _serverName;
-
-        [Parameter(Mandatory = true, HelpMessage = "Name of the Azure Analysis Services server to restart")]
-        [ValidateNotNullOrEmpty]
-        public string Instance { get; set; }
-
-        [Parameter(Mandatory = false)]
-        public SwitchParameter PassThru { get; set; }
-
-        public IAsAzureHttpClient AsAzureHttpClient { get; private set; }
-
-        public ITokenCacheItemProvider TokenCacheItemProvider { get; private set; }
-
-        public RestartAzureAnalysisServer()
-        {
-            AsAzureHttpClient = new AsAzureHttpClient(() => new HttpClient());
-            TokenCacheItemProvider = new TokenCacheItemProvider();
-
-        }
-
-        public RestartAzureAnalysisServer(IAsAzureHttpClient asAzureHttpClient, ITokenCacheItemProvider tokenCacheItemProvider)
-        {
-            AsAzureHttpClient = asAzureHttpClient;
-            TokenCacheItemProvider = tokenCacheItemProvider;
-        }
-
-        protected override IAzureContext DefaultContext
-        {
-            get
-            {
-                // Nothing to do with Azure Resource Management context
-                return null;
-            }
-        }
-
-        protected override string DataCollectionWarning
-        {
-            get
-            {
-                return Resources.ARMDataCollectionMessage;
-            }
-        }
-
-        protected override void BeginProcessing()
+        public override void ExecuteCmdlet()
         {
-            base.BeginProcessing();
-
-            if (AsAzureClientSession.Instance.Profile.Environments.Count == 0)
+            if (!ShouldProcess(Instance, Resources.RestartingAnalysisServicesServer))
             {
-                throw new PSInvalidOperationException(string.Format(Resources.NotLoggedInMessage, ""));
+                return;
             }
 
-            _serverName = Instance;
-            Uri uriResult;
+            var restartEndpoint = string.Format(AsAzureEndpoints.RestartEndpointPathFormat, ServerName);
 
-            // if the user specifies the FQN of the server, then extract the server name out of that.
-            // and set the current context
-            if (Uri.TryCreate(Instance, UriKind.Absolute, out uriResult) && uriResult.Scheme == "asazure")
+            using (var message = AsAzureDataplaneClient.CallPostAsync(restartEndpoint).Result)
```

Comment:
so we don't pass auth token here anymore?

Reply:
All the access token code has been stripped out, and is replaced by ServiceClientCredentials, which are retrieved from the AzureSession's AuthenticationFactory (AsAzureDataplaneCmdletBase.CreateAsAzureDataplaneClient()). These credentials are stored in the AsAzureDataplaneClient, and applied to the http request before it is sent.